dotnet/src/Connectors/Connectors.OpenAI/TextToImage/AzureOpenAITextToImageService.cs

```diff
@@ -3,215 +3,151 @@
-    /// <returns></returns>
-    private async Task<AzureOpenAITextToImageResponse> GetTextToImageResultAsync(string operationId, CancellationToken cancellationToken = default)
+    private static OpenAIClientOptions GetClientOptions(HttpClient? httpClient, string? apiVersion)
     {
-        var operationLocation = this.GetUri(GetImageOperation, operationId);
-
-        var retryCount = 0;
-
-        while (true)
+        OpenAIClientOptions.ServiceVersion version = apiVersion switch
         {
-            if (this._maxRetryCount == retryCount)
-            {
-                throw new KernelException("Reached maximum retry attempts");
-            }
-
-            using var response = await this._core.ExecuteRequestAsync(operationLocation, HttpMethod.Get, null, cancellationToken).ConfigureAwait(false);
-            var responseJson = await response.Content.ReadAsStringWithExceptionMappingAsync().ConfigureAwait(false);
-            var result = OpenAITextToImageClientCore.JsonDeserialize<AzureOpenAITextToImageResponse>(responseJson);
-
-            if (result.Status.Equals(AzureOpenAIImageOperationStatus.Succeeded, StringComparison.OrdinalIgnoreCase))
-            {
-                return result;
-            }
-            else if (this.IsFailedOrCancelled(result.Status))
-            {
-                throw new KernelException($"Azure OpenAI text to image {result.Status}");
-            }
-
-            if (response.Headers.TryGetValues("retry-after", out var afterValues) && long.TryParse(afterValues.FirstOrDefault(), out var after))
-            {
-                await Task.Delay(TimeSpan.FromSeconds(after), cancellationToken).ConfigureAwait(false);
-            }
-
-            // increase retry count
-            retryCount++;
-        }
-    }
+            "2022-12-01" => OpenAIClientOptions.ServiceVersion.V2022_12_01,
+            "2023-05-15" => OpenAIClientOptions.ServiceVersion.V2023_05_15,
+            "2023-06-01-preview" => OpenAIClientOptions.ServiceVersion.V2023_06_01_Preview,
+            "2023-07-01-preview" => OpenAIClientOptions.ServiceVersion.V2023_07_01_Preview,
+            "2023-08-01-preview" => OpenAIClientOptions.ServiceVersion.V2023_08_01_Preview,
+            "2023-12-01-preview" => OpenAIClientOptions.ServiceVersion.V2023_12_01_Preview,
+            _ => OpenAIClientOptions.ServiceVersion.V2023_09_01_Preview
+        };
+
+        var options = new OpenAIClientOptions(version)
+        {
+            Diagnostics = { ApplicationId = HttpHeaderValues.UserAgent }
+        };
 
-    private string GetUri(string operation, params string[] parameters)
-    {
-        var uri = new Azure.Core.RequestUriBuilder();
-        uri.Reset(new Uri(this._endpoint));
-        uri.AppendPath(operation, false);
-        foreach (var parameter in parameters)
+        if (httpClient != null)
         {
-            uri.AppendPath("/" + parameter, false);
+            options.Transport = new HttpClientTransport(httpClient);
```

Comment:
We need to disable default Azure SDK retry policy to have only one source of truth - SK to avoid double retries. 

`options.RetryPolicy = new RetryPolicy(maxRetries: 0); // Disable Azure SDK retry policy if and only if a custom HttpClient is provided.`

Reply:
Thanks, added the suggestion